### PR TITLE
Allow users to override baremetal provisioning images

### PR DESCRIPTION
**Pull request by rabi**
Jira: https://issues.redhat.com/browse/OSPRH-9093

### CODE REVIEW

**Review by fao89 — APPROVED** [review]

File: controllers/dataplane/openstackdataplanenodeset_controller.go [hunk 1]
```diff
@@ -369,13 +369,13 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 			condition.InitReason, condition.InitReason)
 
 		// Set Images
-		if containerImages.OsContainerImage != nil {
+		if containerImages.OsContainerImage != nil && instance.Spec.BaremetalSetTemplate.OSContainerImageURL == "" {
```
Comment (slagle) [inline controllers/dataplane/openstackdataplanenodeset_controller.go:372]: i'm good with the change here to allow overrding the image. related though, does the spec change here actually get saved? i thought updating the spec in the reconcile was a bit of an anti-pattern.

**Review by slagle — APPROVED** [review]

File: controllers/dataplane/openstackdataplanenodeset_controller.go (hunk 1 above)
Comment (rabi) [reply controllers/dataplane/openstackdataplanenodeset_controller.go:372]: Good point.. I think it does get saved, but let's change it to update when creating baremetalset spec.

**Comment by rabi** [comment]
`error: build error: Failed to push image: trying to reuse blob sha256:247c2d03e9487628cb6754ff5385a670df160f7bba36af8fc1f2066e461dc36e at destination: pinging container registry quay.rdoproject.org: Get "https://quay.rdoproject.org/v2/": dial tcp 38.129.56.158:443: i/o timeout`

**Review by fao89 — APPROVED** [review]

**Review by slagle — APPROVED** [review]

**Pull request merged**